### ISSUE TITLE
502 Bad Gateway  -- nginx

### ISSUE THREAD

**Issue by Hacktast1c** [facebook/fbctf#94]
Is anyone getting the same issue? 

I have:

>>  1
sudo apt-get install git
git clone https://github.com/facebook/fbctf
cd fbctf
./extra/provision.sh prod $PWD  (ALSO TRIED `PWD`)

>> 2
cd /etc/nginx/certs
openssl genrsa -des3 -out self-ssl.key 2048
openssl req -new -key self-ssl.key -out self-ssl.csr
cp -v self-ssl.{key,original}
openssl rsa -in self-ssl.original -out self-ssl.key
rm -v self-ssl.original
openssl x509 -req -days 4000 -in self-ssl.csr -signkey self-ssl.key -out self-ssl.crt
mv self-ssl.crt fbctf.csr
mv self-ssl.key fbctf.key

service nginx restart

>> I even tried to stage in DEV Mode.  

>> I have re-created my VM 3 times.

>>>> EFFOR LOG <<<<

Is anyone getting the same issue? 

I have:

>>  1
sudo apt-get install git
git clone https://github.com/facebook/fbctf
cd fbctf
./extra/provision.sh prod $PWD  (ALSO TRIED `PWD`)

>> 2
cd /etc/nginx/certs
openssl genrsa -des3 -out self-ssl.key 2048
openssl req -new -key self-ssl.key -out self-ssl.csr
cp -v self-ssl.{key,original}
openssl rsa -in self-ssl.original -out self-ssl.key
rm -v self-ssl.original
openssl x509 -req -days 4000 -in self-ssl.csr -signkey self-ssl.key -out self-ssl.crt
mv self-ssl.crt fbctf.csr
mv self-ssl.key fbctf.key

service nginx restart

>> I even tried to stage in DEV Mode.  

>> I have re-created my VM 3 times.

Thank you in advance ....

**Comment by mehnomy** [comment]
Hello @Hacktast1c,

Could you please try using my guide:
https://github.com/mehnomy/FBCTF-Ubuntu-Setup

If you still have issues please feel free to reply back here.

Kind Regards,
Byron Fisher

**Comment by Hacktast1c** [comment]
Byron, I am getting an "unable to locate package hhvm" message.

**Comment by mehnomy** [comment]
@Hacktast1c,

Did you follow my guide with a clean Ubuntu 14.04 x64 environment?

Kind Regards,
Byron Fisher

**Comment by Hacktast1c** [comment]
I'm using 32 bit.  I will download the x64 version and try again.  TY

**Comment by no1rscyber** [comment]
I went to 64 bit and ensured the certificates were in the correct folder and that fixed my errors.

**Comment by Hacktast1c** [comment]
@no1rscyber @mehnomy :
   
Thank you for your input. Should have followed my gut in the first place.  I ended up using 64 bit server. Flowtype install failed, however, I was still able to hit the interface. 

Everything seems to work well. Thank you very much for your input. Now adding content. I look forward to extending and contributing. 

Cheers

**Comment by evilchurch** [comment]
i installed it on vagrant VM,but when I browse to the 10.10.10.5 I get an nginx 502 error

**Comment by mehnomy** [comment]
hello @evilchurch,

are you trying to build the Production deployment or the Development deployment?
is the provision script running without any errors?

Kind Regards,
Byron Fisher

**Comment by hackmethod** [comment]
@anyone

I am currently now having this issue.  I was running live (production) for a few days adding challenges, adjusting point vaules, adding files for download, messing with the game itself with how a user would interact, etc.  Then work reared is ugly face and I had to leave it... yesterday I found some time to start messing with it again and I get "502 Bad Gateway".  Anyone have any troubleshooting ideas that I can try, bc I would rather not start all over.  Ive restarted the nginx service, updated the certificates, rebooted the server, and one other thing that doesn't come to mind and I still cant get it up and running.  I can ssh in and see the service running... but I dont know what it could be.  Any help would be greatly appreciated. 

Thanks!
HM
